apps/src/labs/projects/utils.ts

```diff
@@ -0,0 +1,25 @@
+import getScriptData from '@cdo/apps/util/getScriptData';
+
+// Partial definition of the App Options structure, only defining the
+// pieces we need in this component.
+interface PartialAppOptions {
+  level: {
+    isProjectLevel: boolean;
+  };
+  channel: string;
+}
+
+/**
+ * Fetch the ID of the current project from the App Options object
+ * if currently on a standalone project level (i.e. under /projects).
+ *
+ * @returns channel ID if on a standalone project level, null if not.
+ */
+export function getStandaloneProjectId(): string | null {
```

Comment:
Curious how y'all feel about this. I wanted to avoid referencing app options too much, but we need a way to figure out if we are on a project level, and this seemed like the most straightforward way. I've moved the logic out into this util in case we end up revising this.

Reply:
Can we get this information from the call to `/level_data` instead?

Reply:
In general, I'm ok with this for now since we aren't planning to support page switching without reloads for standalone projects. But if we can find a way to do it without app_options that would be preferable

Reply:
Yeah, we could determine `isProjectLevel` from the call to `level_data` (technically that's stored on level `properties`, but I was thinking of modifying the API to return the properties object anyway as we'll need other information there, like hide share and remix). We'd still need to determine the channel ID though - another option would be grab it from the URL directly, which is what we do [here](https://github.com/code-dot-org/code-dot-org/blob/f6aabbbe0449410a9246bee86f9706498733b64e/apps/src/code-studio/initApp/project.js#L1703). This felt a little brittle to me though since we're baking in assumptions of how the URL will be structured

Reply:
Oh yeah I don't love inferring the channel id from the url. I guess we do need the channel id from the backend somehow, because we can't get it based on anything available on the frontend besides the url.

Reply:
Yeah - if we want to avoid relying on app options entirely, another option could be to send channel ID in a separate script tag like we do for /projectbeats?

Reply:
hmm yeah we could but I don't think we necessarily need to right now, given we'd be adding it in addition to the app options we are already sending. Would it be worth adding a comment as to why we are still using app options here, and the path to getting off app options in the future? I could also see us eventually just paring down app options to only be the channel id for a standalone project in the future.